src/wallet/db.cpp

```diff
@@ -359,13 +359,12 @@ void CDBEnv::CheckpointLSN(const std::string& strFile)
 }
 
 
-CDB::CDB(const std::string& strFilename, const char* pszMode, bool fFlushOnCloseIn) : pdb(NULL), activeTxn(NULL)
+CDB::CDB(CWalletDBWrapper& dbw, const char* pszMode, bool fFlushOnCloseIn) : pdb(NULL), activeTxn(NULL)
```

Comment:
CWalletDBWrapper is not a wrapper around CWalletDB. We kind of have a naming problem with the wallet:
- `CDBEnv` is an environment in which the database exists (has no analog in `dbwrapper.h`)
- `CWalletDBWrapper` represents a wallet database (similar to `CDBWrapper` in `dbwrapper.h`)
- `CDB` is a low-level database transaction (similar to `CDBBatch` in `dbwrapper.h`)
- `CWalletDB` is a modifier object for the wallet, and encapsulates a database transaction as well as methods to act on the database (no analog in `dbwrapper.h`)

The latter two are named badly, should be renamed at some point but didn't want to do it here.